### ISSUE TITLE
Error: required link 'created_by' of object type 'default::Item' hidden by access policy

### ISSUE THREAD

**Issue by stemil23** [edgedb/nextjs-edgedb-auth-template#20]
When testing with two (or more) users with each having created ToDo items the dashboard errors. 
I believe the schema lines 40, 41 are the cause. 
```
    access policy everyone_insert_only
      allow insert;
```

As when "select" is added to line 41 the defect is eliminated. 
```
    access policy everyone_insert_only
      allow insert, select;
```

It seem that the Select Query on dashboard/page is the cause of this defect as removing "created by" (lines 17-20) eliminates the error:
```
  const items = await client.query<Props["items"][number]>(`
    select Item {
      id,
      name,
      created,
      updated,
      created_by: {
        name,
        email
      }
    };
  `); 
```

![Cursor_and_localhost_3000_dashboard](https://github.com/user-attachments/assets/9bc50934-6eda-467d-b3ba-fb8ca6cb35d1)
<img width="135" alt="EdgeDB___EdgeDB_Auth___Next_js_Template" src="https://github.com/user-attachments/assets/b6f3a709-1b48-4486-8494-e00729a80eea">
![nextjs_edgedb_auth_template_main___main___EdgeDB_Local-2](https://github.com/user-attachments/assets/e140d227-bd8b-4b75-ad02-c132ec51a39a)
![nextjs_edgedb_auth_template_main___main___EdgeDB_Local](https://github.com/user-attachments/assets/f9782988-87c6-4f68-b849-1987c208e0eb)

**Comment by bysnet** [comment]
> When testing with two (or more) users with each having created ToDo items the dashboard errors. I believe the schema lines 40, 41 are the cause.
> 
> ```
>     access policy everyone_insert_only
>       allow insert;
> ```
> 
> As when "select" is added to line 41 the defect is eliminated.
> 
> ```
>     access policy everyone_insert_only
>       allow insert, select;
> ```
> 
> It seem that the Select Query on dashboard/page is the cause of this defect as removing "created by" (lines 17-20) eliminates the error:
> 
> ```
>   const items = await client.query<Props["items"][number]>(`
>     select Item {
>       id,
>       name,
>       created,
>       updated,
>       created_by: {
>         name,
>         email
>       }
>     };
>   `); 
> ```
> 
> ![Cursor_and_localhost_3000_dashboard](https://private-user-images.githubusercontent.com/1161759/378124335-9bc50934-6eda-467d-b3ba-fb8ca6cb35d1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM4MTcxOTksIm5iZiI6MTczMzgxNjg5OSwicGF0aCI6Ii8xMTYxNzU5LzM3ODEyNDMzNS05YmM1MDkzNC02ZWRhLTQ2N2QtYjNiYS1mYjhjYTZjYjM1ZDEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIxMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMTBUMDc0ODE5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MTJjNWJkMDk2OThiZjFjZDdkZGFhMDQ5YTc2YjgyNzFiNzA5ZGNkMDA3ZGRhZDlmMzUwNTJkMTAyYjA3NjczMyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.656aDN5wJtKCGJBShWwNF0TbMFVt5H6OKMpyRee-0Ik) <img alt="EdgeDB___EdgeDB_Auth___Next_js_Template" width="135" src="https://private-user-images.githubusercontent.com/1161759/378124337-b6f3a709-1b48-4486-8494-e00729a80eea.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM4MTcxOTksIm5iZiI6MTczMzgxNjg5OSwicGF0aCI6Ii8xMTYxNzU5LzM3ODEyNDMzNy1iNmYzYTcwOS0xYjQ4LTQ0ODYtODQ5NC1lMDA3MjlhODBlZWEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIxMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMTBUMDc0ODE5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ODY2ZDA0Y2FiNTQ1NzU0ZjJhMWM5NDMwNWJhZThlMmUzZTg0NWFhMDU0ZThiNzM4ZTdmM2QyZGM1ZjY3YThiOSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.s8BpEkWFplU-kj8ViwmdxXmweINS93_aRiS5d_-ZjcI"> ![nextjs_edgedb_auth_template_main___main___EdgeDB_Local-2](https://private-user-images.githubusercontent.com/1161759/378124339-e140d227-bd8b-4b75-ad02-c132ec51a39a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM4MTcxOTksIm5iZiI6MTczMzgxNjg5OSwicGF0aCI6Ii8xMTYxNzU5LzM3ODEyNDMzOS1lMTQwZDIyNy1iZDhiLTRiNzUtYWQwMi1jMTMyZWM1MWEzOWEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIxMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMTBUMDc0ODE5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NDMzZjMzNGM0YzFjM2Y1OTc2YWY1NWI0MzRmZGMwY2M5YWE3YTk3OWQwNzk2ZTMzMTExMjQ4ODgzYjk3ODZmMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.5W-AA3oMU6dmI1TDr-WQgXhkDkZDM9KYOgTP7Cm7ECA) ![nextjs_edgedb_auth_template_main___main___EdgeDB_Local](https://private-user-images.githubusercontent.com/1161759/378124340-f9782988-87c6-4f68-b849-1987c208e0eb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM4MTcxOTksIm5iZiI6MTczMzgxNjg5OSwicGF0aCI6Ii8xMTYxNzU5LzM3ODEyNDM0MC1mOTc4Mjk4OC04N2M2LTRmNjgtYjg0OS0xOTg3YzIwOGUwZWIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIxMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMTBUMDc0ODE5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9Y2IwMWNlODM2YWE1NTcxZWE2NzUwNDViMzBmMWRlOTYyNDk2NDFmNTZhNGQ3MTQwYmIzMzQxOTdmY2M1ODRmNCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.A4_2fULEEQqP5uZwC2Mlxvk13T779Ox0Gh-1nK7sUic)

**Comment by lonexw** [comment]
yes, it's a problem, The user's access policy should indeed be slightly modified.
